### PR TITLE
make sure wells has valid controls

### PR DESCRIPTION
**Pull request by totto82**
This change the logic of the well control copying between episodes

The new logic 
1) Always initialize with valid control from eclipeState
2) Copy control from the last time step for existing wells. 

Note that the new logic is consistent with the rest of the initialization of the well state.

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
benchmark please

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test    | Configuration                                     | Relative |
| ------- | ------------------------------------------------- | -------- |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 1        | 0.979    |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 8        | 0.988    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 1        | 1.027    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 8        | 1.034    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 1 | 1.002    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 | 1.001    |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=1101

**Comment by totto82** [comment]
The   ` mpi.compareECLFiles_flow+9_EDITNNC_MODEL2`  case showed significant differences. The last commit tries to improve this by improving the initialization of WAG injectors.

**Comment by totto82** [comment]
benchmark opm-common=2335 please

**Comment by totto82** [comment]
benchmark please

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test    | Configuration                                     | Relative |
| ------- | ------------------------------------------------- | -------- |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 1        | 1.017    |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 8        | 0.98     |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 1        | 1.035    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 8        | 1.043    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 1 | 1.002    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 | 1.003    |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=1105

**Comment by totto82** [comment]
I would very much like this to go in as soon as possible as it fixes an urgent issue on an important model by avoiding uninitialized well controls. With the latest fix for WAG injectors  "Benchmark" does not show significant performance hit. 
I have also manually checked the Jenkins failures. There are differences but IMO not significant anymore.

**Comment by joakim-hove** [comment]
jenkins build this update_data please

File: opm/simulators/wells/WellStateFullyImplicitBlackoil.hpp [hunk 1]
```diff
@@ -207,12 +222,8 @@ namespace Opm
                          //   continue;
                          //}
 
-                        // if there is no effective control event happens to the well, we use the current_injection/production_controls_ from prevState
-                        // otherwise, we use the control specified in the deck
-                        if (!effective_events_occurred_[w]) {
-                            current_injection_controls_[ newIndex ] = prevState->currentInjectionControls()[ oldIndex ];
-                            current_production_controls_[ newIndex ] = prevState->currentProductionControls()[ oldIndex ];
-                        }
+                        current_injection_controls_[ newIndex ] = prevState->currentInjectionControls()[ oldIndex ];
```
Comment (joakim-hove) [inline opm/simulators/wells/WellStateFullyImplicitBlackoil.hpp:225]: There is some old commented out code directly above this which should be removed.

**Comment by joakim-hove** [comment]
OK;

I have looked at this - and to me it looks correct; I would have thought that it should be identical to master - so why are there changes at all? Is that expected?

**Comment by totto82** [comment]
> Is that expected?

Yes. The logic of when the last time steps control is used and when the control given by the deck is used is changed. Now we use (re)set the time step control from the deck if 1) we have a new well 2) the well change from injector to producer. For WAG injectors (relevant for many of the failed tests) I added a extra well solve to get better initial values.

**Comment by joakim-hove** [comment]
jenkins build this opm-tests=502 please

**Pull request merged**